### PR TITLE
[feat] - Add Labels to RuleGroup

### PR DESCRIPTION
**Pull request by heliapb**
## Description

Fixes issue https://github.com/prometheus-operator/prometheus-operator/issues/7135

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Verification

Please check the [Prometheus-Operator testing guidelines](../TESTING.md) for recommendations about automated tests.

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

File: pkg/apis/monitoring/v1/prometheusrule_types.go [hunk 1]
```diff
@@ -63,6 +63,8 @@ type RuleGroup struct {
 	// Name of the rule group.
 	// +kubebuilder:validation:MinLength=1
 	Name string `json:"name"`
+	// Labels to add or overwrite before storing the result for its rules.
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/prometheusrule_types.go:66]:
```suggestion
	// Labels to add or overwrite before storing the result for its rules.
	// The labels defined at the rule level take precedence.
	//
	// +optional
```

**Review by simonpasquier — COMMENTED** [review]
Thanks for starting this! The key point is to check for version compatibility.

File: pkg/apis/monitoring/v1/prometheusrule_types.go [hunk 2]
```diff
@@ -63,6 +63,11 @@ type RuleGroup struct {
...
+	// The labels defined at the rule level take precedence.
+	//
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1/prometheusrule_types.go:68]:
```suggestion
	//
	// It requires Prometheus >= 3.0.0.
	// The field is ignored for Thanos Ruler.
```

**Comment by heliapb** [comment]
Hi @simonpasquier sorry for the delay, and thanks for the feedback. Pls review

File: pkg/operator/rules.go [hunk 3]
```diff
@@ -110,6 +110,7 @@ func (prs *PrometheusRuleSelector) sanitizePrometheusRulesSpec(promRuleSpec moni
 	minVersionKeepFiringFor := semver.MustParse("2.42.0")
 	minVersionLimits := semver.MustParse("2.31.0")
 	minVersionQueryOffset := semver.MustParse("2.53.0")
+	minVersionRuleGroupLabels := semver.MustParse("3.0.0")
```
Comment (simonpasquier) [inline pkg/operator/rules.go:113]: I don't think that Thanos supports this yet? We need to do the same as for minVersionQueryOffset (see L119 below).

**Pull request merged**